localstack/services/logs/provider.py

```diff
@@ -250,8 +250,12 @@ def moto_put_subscription_filter(fn, self, *args, **kwargs):
     if not log_group:
         raise ResourceNotFoundException("The specified log group does not exist.")
 
+    arn_data = parse_arn(destination_arn)
+
     if ":lambda:" in destination_arn:
-        client = connect_to(region_name=extract_region_from_arn(destination_arn)).lambda_
+        client = connect_to(
+            aws_access_key_id=arn_data["account"], region_name=arn_data["region"]
+        ).lambda_
```

Comment:
I believe this might be the same issue we had in S3 notifications. If `logs` also cross-account/cross-region destinations, I don't believe we need to add the account here to the client call, as lambda is already cross-account/cross-region if you use the lambda function ARN. 
You should use the ARN here in `client.get_function(FunctionName=destination_arn)` instead.
This will help later on with IAM enforcement I think. We can then remove the `lambda_name = arns.lambda_function_name(destination_arn)` line. 
This is how lambda uses the ARN:
```python
    def get_function(
        self,
        context: RequestContext,
        function_name: NamespacedFunctionName,
        qualifier: Qualifier = None,
    ) -> GetFunctionResponse:
        account_id, region = api_utils.get_account_and_region(function_name, context)
        function_name, qualifier = api_utils.get_name_and_qualifier(
            function_name, qualifier, context
        )
```